uber/models.py

```diff
@@ -1094,9 +1094,11 @@ class Attendee(MagModel, TakesPaymentMixin):
     dept_checklist_items = relationship('DeptChecklistItem', backref='attendee')
 
     _repr_attr_names = ['full_name']
-    __table_args__ = (
-        UniqueConstraint('badge_num', deferrable=True, initially='DEFERRED'),
-    )
+
+    if Session.engine.dialect == 'postgresql':
```

Comment:
This isn't quite right, which is to say that it's "fixing" the tests in a way you don't intend.

This if-statement is executed once when the module is first loaded.  That means that when we monkeypatch our ``Session.engine`` object when the tests are initializing, it won't go back and modify ``Attendee.__table_args__`` because that was set at import time.

So why does this fix the tests?  In fact it's because as written the if-statement is always false!  It's because ``Session.engine.dialect`` isn't actually a string:

```python
>>> Session.engine.dialect
<sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7f33f0e74eb8>
>>> Session.engine.dialect == 'postgresql'
False
```

You need to refer to ``Session.engine.dialect.name`` to get what you want:

```python
>>> Session.engine.dialect.name
'postgresql'
>>> Session.engine.dialect.name == 'postgresql'
True
```

But if you do that then you'll probably go back to having the unit tests fail.  But why?

The answer is that in our unit tests we change over from Postgres to SQLite at
https://github.com/magfest/ubersystem/blob/master/uber/tests/conftest.py#L74
but we've already set ``Attendee.__table_args__`` when that code is executed to something that won't work with SQLite.  I believe that if we just say

```python
Attendee.__table_args__ = ()
```

in the ``init_db`` function then it should fix the problem, and then we can go back to just setting ``__table_args__`` without wrapping it in an if-statement.